### ISSUE TITLE
SQLite database lock when running buildbot create-master with pypy

### ISSUE THREAD

**Issue by rvanlaar** [buildbot/buildbot#6478]
```
roland@MiX ~/wb23> poetry run buildbot create-master master
mkdir /home/roland/wb23/master
creating /home/roland/wb23/master/master.cfg.sample
creating database (sqlite:///state.sqlite)
retrying <function Model.upgrade.<locals>.thd at 0x00007fc08e7ea7a0> after sql error (_sqlite3.OperationalError) database is locked
[SQL: 
CREATE TABLE test_unicode (
	u VARCHAR(100), 
	b BLOB
)
```

**Steps to reproduce:**
- install poetry: https://python-poetry.org/docs/
- Download latest 3.8 pypy: https://www.pypy.org/download.html
- mkdir tmp && cd tmp
- put the pyproject.toml file in the in new tmp directory
- poetry env use /path/to/pypy/bin/python3
- poetry install
- poetry run buildbot create-master master
ERROR

Ubuntu 21.04
pypy3.8-v7.3.6

It does work when using the system python.

pyproject.toml
```
[tool.poetry]
name = "crash"
version = "0.1.0"
description = ""
authors = ["Roland <roland@example.com>"]

[tool.poetry.dependencies]
python = "^3.8"
buildbot = {extras = ["bundle"], version = "^3.5.0"}

[tool.poetry.dev-dependencies]

[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"
```

Full log:
```
mkdir /home/roland/wb23/master
creating /home/roland/wb23/master/master.cfg.sample
creating database (sqlite:///state.sqlite)
retrying <function Model.upgrade.<locals>.thd at 0x00007fc08e7ea7a0> after sql error (_sqlite3.OperationalError) database is locked
[SQL: 
CREATE TABLE test_unicode (
	u VARCHAR(100), 
	b BLOB
)

]
(Background on this error at: https://sqlalche.me/e/14/e3q8)
Traceback (most recent call last):
  File "/home/roland/wb23/.venv/lib/pypy3.8/site-packages/twisted/python/threadpool.py", line 261, in <lambda>
    ctx, func, *args, **kw
  File "/home/roland/wb23/.venv/lib/pypy3.8/site-packages/twisted/python/context.py", line 117, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/home/roland/wb23/.venv/lib/pypy3.8/site-packages/twisted/python/context.py", line 82, in callWithContext
    return func(*args, **kw)
  File "/home/roland/wb23/.venv/lib/pypy3.8/site-packages/buildbot/db/pool.py", line 226, in __thd
    log.err(e, f'retrying {callable} after sql error {e}')
--- <exception caught here> ---
  File "/home/roland/wb23/.venv/lib/pypy3.8/site-packages/buildbot/db/pool.py", line 207, in __thd
    rv = callable(arg, *args, **kwargs)
  File "/home/roland/wb23/.venv/lib/pypy3.8/site-packages/buildbot/db/model.py", line 1082, in thd
    test_unicode(conn)
  File "/home/roland/wb23/.venv/lib/pypy3.8/site-packages/buildbot/db/migrate_utils.py", line 33, in test_unicode
    test_unicode.create()
  File "/home/roland/wb23/.venv/lib/pypy3.8/site-packages/sqlalchemy/sql/schema.py", line 959, in create
    bind._run_ddl_visitor(ddl.SchemaGenerator, self, checkfirst=checkfirst)
  File "/home/roland/wb23/.venv/lib/pypy3.8/site-packages/sqlalchemy/engine/base.py", line 2130, in _run_ddl_visitor
    visitorcallable(self.dialect, self, **kwargs).traverse_single(element)
  File "/home/roland/wb23/.venv/lib/pypy3.8/site-packages/sqlalchemy/sql/visitors.py", line 524, in traverse_single
    return meth(obj, **kw)
  File "/home/roland/wb23/.venv/lib/pypy3.8/site-packages/sqlalchemy/sql/ddl.py", line 898, in visit_table
    include_foreign_key_constraints,  # noqa
  File "/home/roland/wb23/.venv/lib/pypy3.8/site-packages/sqlalchemy/engine/base.py", line 1306, in execute
    return meth(self, multiparams, params, _EMPTY_EXECUTION_OPTS)
  File "/home/roland/wb23/.venv/lib/pypy3.8/site-packages/sqlalchemy/sql/ddl.py", line 81, in _execute_on_connection
    self, multiparams, params, execution_options
  File "/home/roland/wb23/.venv/lib/pypy3.8/site-packages/sqlalchemy/engine/base.py", line 1404, in _execute_ddl
    compiled,
  File "/home/roland/wb23/.venv/lib/pypy3.8/site-packages/sqlalchemy/engine/base.py", line 1863, in _execute_context
    e, statement, parameters, cursor, context
  File "/home/roland/wb23/.venv/lib/pypy3.8/site-packages/sqlalchemy/engine/base.py", line 2044, in _handle_dbapi_exception
    sqlalchemy_exception, with_traceback=exc_info[2], from_=e
  File "/home/roland/wb23/.venv/lib/pypy3.8/site-packages/sqlalchemy/util/compat.py", line 207, in raise_
    raise exception
  File "/home/roland/wb23/.venv/lib/pypy3.8/site-packages/sqlalchemy/engine/base.py", line 1820, in _execute_context
    cursor, statement, parameters, context
  File "/home/roland/wb23/.venv/lib/pypy3.8/site-packages/sqlalchemy/engine/default.py", line 732, in do_execute
    cursor.execute(statement, parameters)
  File "/home/roland/projects/scummvm-director/buildbot/pypy/pypy3.8-v7.3.6-linux64/lib/pypy3.8/_sqlite3.py", line 838, in wrapper
    return func(self, *args, **kwargs)
  File "/home/roland/projects/scummvm-director/buildbot/pypy/pypy3.8-v7.3.6-linux64/lib/pypy3.8/_sqlite3.py", line 999, in execute
    return self.__execute(False, sql, [params])
  File "/home/roland/projects/scummvm-director/buildbot/pypy/pypy3.8-v7.3.6-linux64/lib/pypy3.8/_sqlite3.py", line 947, in __execute
    self.__statement = self.__connection._statement_cache.get(sql)
  File "/home/roland/projects/scummvm-director/buildbot/pypy/pypy3.8-v7.3.6-linux64/lib/pypy3.8/_sqlite3.py", line 192, in get
    stat = Statement(self.connection, sql)
  File "/home/roland/projects/scummvm-director/buildbot/pypy/pypy3.8-v7.3.6-linux64/lib/pypy3.8/_sqlite3.py", line 1157, in __init__
    raise self.__con._get_exception(ret)
sqlalchemy.exc.OperationalError: (_sqlite3.OperationalError) database is locked
[SQL: 
CREATE TABLE test_unicode (
	u VARCHAR(100), 
	b BLOB
)

]
(Background on this error at: https://sqlalche.me/e/14/e3q8)
```

**Comment by tdesveaux** [comment]
Could reproduce with pypy 3.10 and Buildbot 3.11.
Issue seem fixed with latest 4.0.1

**Comment by p12tic** [comment]
Closing as fixed. Please reopen if this happens in Buildbot 4.x
